src/Illuminate/Routing/RouteCollection.php

```diff
@@ -193,11 +206,27 @@ protected function matchAgainstRoutes(array $routes, $request, $includingMethod
             return $route->isFallback;
         });
 
-        return $routes->merge($fallbacks)->first(function ($value) use ($request, $includingMethod) {
-            return $value->matches($request, $includingMethod);
+        return $routes->merge($fallbacks)->first(function (Route $route) use ($request, $includingMethod) {
+            return $route->matches($request, $includingMethod);
         });
     }
 
+    /**
+     * Determine if a route in the compiled array matches the request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Routing\Route|null
+     */
+    protected function matchAgainstCompiledRoutes($request)
+    {
+        $context = (new RequestContext())->fromRequest($request);
+        $matcher = new CompiledUrlMatcher($this->compiledRoutes, $context);
```

Comment:
This is the new Symfony `CompiledUrlMatcher` which should be the fastest way possible of matching routes. However, it doesn't seems to do much difference with the current system...